packages/gatsby/src/utils/get-public-path.js

```diff
@@ -0,0 +1,21 @@
+const trimSlashes = part => part.replace(/(^\/)|(\/$)/g, ``)
+
+const isURL = possibleUrl =>
+  [`http`, `//`].some(expr => possibleUrl.startsWith(expr))
```

Comment:
What is this fixing though? It seems like `http` would also catch `https`?

Reply:
Just with `http` there is also a `httpfolder/` a valid match

Reply:
@muescha do you have a practical example of what that exactly means?

Reply:
In other words--when would a user run into that? Or are you saying, in theory, this is too leniant?

Reply:
Yes. There some edge cases where the function returns wrong result. If the function means isUrl then it should be an Url ;)

`http//domain` is true
`httpfolder` is true

Reply:
Also as I see it right from the code:

It never hit the `//` check

Because a `//something` is always stripped with `stripSlashes` to `/something`

Reply:
https://github.com/gatsbyjs/gatsby/blob/5521302fb42edb88a445016aed4445a0c423a56a/packages/gatsby/src/utils/__tests__/get-public-path.js#L58

Reply:
But yeah - agreed, we can make this more correct easily.

Reply:
Ok. `handles double slashes` that's magic ;)

Reply:
Ok - I see - because the `//` check is never `true` in `isUrl` for this it added later the `/` to the `/something` again to be again an `//something`